mkosi/distributions/opensuse.py

```diff
@@ -28,6 +28,43 @@ def initrd_path(kver: str) -> Path:
         return Path("boot") / f"initrd-{kver}"
 
 
+def zypper_addrepo(state, url, name, caching=False):
+    run(["zypper", "--root", state.root, "addrepo", "-c", "-k" if caching else "-K", url, name])
+
+
+def zypper_removerepo(state, repo):
+    run(["zypper", "--root", state.root, "removerepo", repo])
+
+
+def zypper_modifyrepo(state, repo, caching=None):
+    cmdline = ["zypper", "--root", state.root, "modifyrepo"]
+
+    if caching is not None:
+        cmdline += ["-k" if caching else "-K"]
+
+    run(cmdline + [repo])
+
+
+def zypper_install(state, packages):
+    if not state.config.with_docs:
+        # zypper has no option for excluding the docs...
+        state.root.joinpath("etc/zypp/zypp.conf").write_text("rpm.install.excludedocs = yes\n")
```

Comment:
Can we tell zypper to look for its configuration elsewhere? If so, I'd prefer if we could write it under `state.workspace` to avoid touching the image as little as possible.

Reply:
I think in this case it's better to make the configuration of zypper used to create the rootfs persistent so subsequent calls to zypper behave consistently. It would be odd for example if the docs was excluded initially but suddenly installed when the image is updated.

Reply:
Current policy for mkosi is to not configure the package manager in the image, but just use the distribution provided config there and leave it to the user to configure it in the image itself. We already do this for the other distros, so we should do the same for opensuse.

We tried to support updating images in the past but it turns out its a huge can of worms so we're moving more towards the "golden image" usecase where we don't expect images to be updated in the traditional way after we've produced. If users want an image to behave in that way, it's possible but they'll have to do the configuration required for that themselves.

Reply:
Even if not used later, the conf file can be still used to document the way the image was setup though. 

BTW my patch isn't changing the location of the configuration file, it was done this way previously.

If you think that needs to be changed, we can do that later in a different PR.